crates/ty/src/lib.rs

```diff
@@ -288,23 +294,41 @@ impl MainLoop {
                             let mut max_severity = Severity::Info;
                             let diagnostics_count = result.len();
 
-                            for diagnostic in result {
-                                write!(
+                            if terminal_settings.output_format
+                                == ruff_db::diagnostic::DiagnosticFormat::Json
```

Comment:
Can you explain why Json needs special handling compared to the rest of the diagnostic rendering?